WebApp/Controllers/HomeController.cs

```diff
@@ -51,22 +48,31 @@ public ActionResult About()
         public async Task<ActionResult> SendMail()
         {            
             // try to get token silently
+            IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
+                .Create(clientId)
+                .WithRedirectUri(redirectUri)
+                .WithClientSecret(appKey)
+                .Build();
+
             string signedInUserID = ClaimsPrincipal.Current.FindFirst(ClaimTypes.NameIdentifier).Value;
-            TokenCache userTokenCache = new MSALSessionCache(signedInUserID, this.HttpContext).GetMsalCacheInstance();            
-            ConfidentialClientApplication cca = new ConfidentialClientApplication(clientId, redirectUri,new ClientCredential(appKey), userTokenCache, null);
+            new MSALSessionCache(cca.UserTokenCache, signedInUserID, HttpContext);
```

Comment:
Not a big fan of this construct. The ctor is supposed to create objects, but in this case we're just invoking a constructor that does some work to hook up the cache. A static method would be better.